src/ConnectedMode.UnitTests/BranchMatcherTests.cs

```diff
@@ -0,0 +1,211 @@
+            var commit8 = new CommitWrapper(8);
+            var commit9 = new CommitWrapper(9);
+            var commit10 = new CommitWrapper(10);
+            
+            var masterBranch = CreateBranch("master", commit10, commit9, commit8, commit3, commit2, commit1);
+            var branch1 = CreateBranch("branch1", commit5, commit4, commit3, commit2, commit1);
+            var branch2 = CreateBranch("branch2", commit6, commit3, commit2, commit1);
+            var branch3 = CreateBranch("branch3", commit7, commit3, commit2, commit1);
+
+            var repo = CreateRepo(branch3, masterBranch, branch1, branch2);
+
+            var testSubject = new BranchMatcher(service, repo);
+
+            var result = await testSubject.GetMatchedBranch("projectKey");
+
+            result.Should().Be("branch2");
+        }
+
+        [TestMethod]
+        public async Task GetMatchedBranch_NoMatchingBranch_ChooseMain()
+        {
+            var service = CreateService("master", "branch1", "branch2");
+
+            var masterCommit = new CommitWrapper(1);
+            var branch1Commit = new CommitWrapper(2);
+            var branch2Commit = new CommitWrapper(3);
+            var headCommit = new CommitWrapper(4);
+
+            var masterBranch = CreateBranch("master", masterCommit);
+            var branch1 = CreateBranch("branch1", branch1Commit, masterCommit);
+            var branch2 = CreateBranch("branch2", branch2Commit, masterCommit);
+            var headBranch = CreateBranch("head", headCommit);
+
+            var repo = CreateRepo(headBranch, masterBranch, branch1, branch2);
+
+            var testSubject = new BranchMatcher(service, repo);
+
+            var result = await testSubject.GetMatchedBranch("projectKey");
+
+            result.Should().Be("master");
+        }
+
+        [TestMethod]
+        public async Task GetMatchedBranch_RepoHasNoHead_ReturnsNull()
+        {
+            var testSubject = new BranchMatcher(Mock.Of<ISonarQubeService>(), Mock.Of<IRepository>());
+
+            var result = await testSubject.GetMatchedBranch("projectKey");
+
+            result.Should().BeNull();
+        }
+
+        private static IRepository CreateRepo(Branch headBranch, params Branch[] branches)
+        {
+            var branchList = new List<Branch>();
+            branchList.Add(headBranch);
+            branchList.AddRange(branches);
+
+            var allBranches = new BranchCollectionWrapper(branchList);
+            
+            var repo = new Mock<IRepository>();
+
+            repo.SetupGet(r => r.Head).Returns(headBranch);
+            repo.SetupGet(r => r.Branches).Returns(allBranches);
+            return repo.Object;
+        }
+
+        //Order of the commits matter. Commits should be in descending order. 
+        private static BranchWrapper CreateBranch(string branchName, params CommitWrapper[] commits)
+        {
+            var masterBranch = new BranchWrapper(branchName, new CommitLogWrapper(commits));
```

Comment:
This is just `branch`, isn't it? i.e. there's nothing that makes it a `master` branch?

Reply:
yes, we do not care what's master/main on git, we only care about on SQ/SC

Reply:
So calling the variable `masterBranch` is misleading. Could you change it please?